proposals/0002-Extracted-Command-Line-Tools.md

```diff
@@ -0,0 +1,69 @@
+
+This proposal suggests no changes to the end user in its first iteration. We would simply couple new package with React Native version, just like we do with Metro. That makes it possible to still make React Native version specific changes to the CLI itself. 
+
+## Motivation
+
+Facebook does not use CLI provided with React Native, which makes it hard to maintain the project. There is no clear owner and issues / pull requests get stack in the sea of other, non-CLI related reports.
+
+It is also hard to keep moving on the CLI as every additional dependency or code change is subject to the heavy review process and sync.
+
+Having a separate package with me (and Callstack) being strategic maintainers would allow us to iterate on the design of CLI even more and make sure we have the best tooling out there.
+
+Now, if you have been following React Native development for a few years, you know I have been always super excited about CLI in general. I did many talks in that field and lead efforts of creating RNPM and then, merging it to the core as `react-native link` together with Alexey Kureev. Link, and RNPM, in general, was just the first step on our way to meet that high expectation of high-quality tools. Unfortunately, due to the aforementioned constraints and lack of time we were never able to complete our long-term strategy.
+
+I would like to take the slimmering process as an opportunity to revisit that effort and start working on it once again, together with the community.
+
+## Detailed design
+
+For the first release, the goal is to just extract out the CLI into a separate package. Since every command is a standalone module, we would like to use Yarn workspaces to build a modular architecture around it.
+
+Not many people know, but not only CLI allows you to provide your custom commands but also to override the built-in ones. That makes it even more obvious to create a package-oriented setup. The similar setup is already followed by React and Metro and we would take learnings from there.
+
+Apart from reorganizing the architecture, we would like to create a website that describes the features present in the CLI - many of them, making it much easier to work at a scale. Things like support for different locations of iOS/Android projects, linking assets from 3rd party modules are great features that only selected people were fortunate to learn about.
+
+As a next step, we would like to focus on reinvestigating every single piece of CLI to see what we can do better these days. That includes looking once again into the design of the scaffolding system (that is great, but has its limitations), lock-in to Metro (because why it should be the forced default?) and rewriting link to support modern languages and new module system. 
```

Comment:
> rewriting link to support modern languages and new module system. 

This seems like it could be its whole own proposal... 😨 

What do you think about keeping the proposal limited to just the extraction of the CLI for now?

Reply:
Yes, I gone too deep with my plans for the CLI in general. I will discuss them today on the meeting and get some initial ideas, but for now, I am going to remove 50% of this proposal and just make it about extracting from the repository itself.